src/compiler/checker.ts

```diff
@@ -13443,7 +13443,7 @@ namespace ts {
         }
 
         function getTypeOfFirstParameterOfSignature(signature: Signature) {
-            return getTypeAtPosition(signature, 0);
+            return getTypeWithFacts(getTypeAtPosition(signature, 0), TypeFacts.NEUndefined);

```

Comment:
what about cases like this:
```ts
declare function resolve1<T>(value: T): Promise<T>;
declare function resolve2<T>(value: T): Windows.Foundation.IPromise<T>;

async function f(x?: number) {
    let x1 = await resolve1(x);
    let x2 = await resolve2(x);
}
```
Currently type of `x1` is `number | undefined` and `x2` fails with error addressed by this PR.
With the fix type of `x2` will be just `number` which will be inconsistent with type for `x1`